sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UseStringIsNullOrEmpty.cs

```diff
@@ -51,34 +50,25 @@ protected override void Initialize(SonarAnalysisContext context)
                 {
                     var invocationExpression = (InvocationExpressionSyntax)c.Node;
 
-                    if (!(invocationExpression.Expression is MemberAccessExpressionSyntax memberAccessExpression))
+                    if (invocationExpression.Expression is MemberAccessExpressionSyntax memberAccessExpression &&
+                        TryGetFirstArgument(invocationExpression, out var firstArgument) &&
+                        memberAccessExpression.ToStringContains(EqualsName) &&
```

Comment:
I would swap these to lines for code semantic reasons. First check if it's the member we're interested in and then deal with it's arguments.
```suggestion
                        memberAccessExpression.ToStringContains(EqualsName) &&
                        TryGetFirstArgument(invocationExpression, out var firstArgument) &&

```